lib/LaTeXML/Engine/plain.pool.ltxml

```diff
@@ -237,36 +207,20 @@ DefPrimitive('\wlog{}', sub {
     return; },
   locked => 1);
 # From plain.tex
-DefPrimitive('\newcount  DefToken', sub {
-    DefRegisterI($_[1], undef, Number(0), allocate => '\count'); });
-DefPrimitive('\newdimen  DefToken', sub {
-    DefRegisterI($_[1], undef, Dimension(0), allocate => '\dimen'); });
-DefPrimitive('\newskip   DefToken', sub {
-    DefRegisterI($_[1], undef, Glue(0), allocate => '\skip'); });
-DefPrimitive('\newmuskip DefToken', sub {
-    DefRegisterI($_[1], undef, MuGlue(0), allocate => '\muskip'); });
-AssignValue(allocated_boxes => 0);
-DefPrimitive('\newbox DefToken', sub {
-    my $n = LookupValue('allocated_boxes');
-    AssignValue(allocated_boxes => $n + 1, 'global');
-    AssignValue("box$n", List());
-    DefRegisterI($_[1], undef, Number($n), readonly => 1); });
-DefPrimitive('\newhelp DefToken {}', sub { AssignValue(ToString($_[1]) => $_[2]); });
-DefPrimitive('\newtoks DefToken',    sub { DefRegisterI($_[1], undef, Tokens()); });
-# the next 4 actually work by doing a \chardef instead of \countdef, etc.
-# which means they actually work quite differently
-DefPrimitive('\alloc@@ {}', sub {
-    my ($stomach, $type) = @_;
-    my $c = 'allocation @' . ToString($type);
-    my $n = LookupValue($c) || '0';
-    $n = $n->valueOf if ref $n;
-    AssignValue($c => $n + 1, 'global');
-    AssignRegister('\allocationnumber' => Number($n), 'global'); });
-DefMacro('\newread DefToken',     '\alloc@@{read}\global\chardef#1=\allocationnumber');
-DefMacro('\newwrite DefToken',    '\alloc@@{write}\global\chardef#1=\allocationnumber');
-DefMacro('\newfam DefToken',      '\alloc@@{fam}\global\chardef#1=\allocationnumber');
-DefMacro('\newlanguage DefToken', '\alloc@@{language}\global\chardef#1=\allocationnumber');
-
+RawTeX(<<'EoTeX');
+\outer\def\newcount{\alloc@0\count\countdef\insc@unt}
+\outer\def\newdimen{\alloc@1\dimen\dimendef\insc@unt}
+\outer\def\newskip{\alloc@2\skip\skipdef\insc@unt}
+\outer\def\newmuskip{\alloc@3\muskip\muskipdef\@cclv}
+\outer\def\newbox{\alloc@4\box\chardef\insc@unt}
+%%\let\newtoks=\relax % we do this to allow plain.tex to be read in twice
+\outer\def\newhelp#1#2{\newtoks#1#1\expandafter{\csname#2\endcsname}}
+\outer\def\newtoks{\alloc@5\toks\toksdef\@cclv}
+\outer\def\newread{\alloc@6\read\chardef\sixt@@n}
+\outer\def\newwrite{\alloc@7\write\chardef\sixt@@n}
+\outer\def\newfam{\alloc@8\fam\chardef\sixt@@n}
+\outer\def\newlanguage{\alloc@9\language\chardef\@cclvi}
+EoTeX
```

Comment:
Dropping Perl definitions in favor of RawTeX? Scary. Hopefully more reliable, but something to keep an eye on while testing...

Reply:
Yeah, scary, but really the underlying theme: most of `plain.pool.ltxml` could disappear if (some incarnation of) `plain.tex` can be read in.